### DISCUSSION TITLE
Project Planning

### DISCUSSION THREAD

**Discussion by alcohen** [VolksEEG/VolksEEG#9] [General]
Typically I start a project with a project plan. Usually a pretty detailed one.

Because of the nature of VolksEEG development right now - we're volunteers, and this first phase is a basically a proof of concept prototype - a detailed plan might not be appropriate. However, I think that something higher level showing general tasks, dependencies and milestones would be helpful.

I've attached the start of something that I think would be appropriate. 

[VolksEEG sample project plan.pdf](https://github.com/VolksEEG/VolksEEG/files/7492887/VolksEEG.sample.project.plan.pdf)

Determining task duration will be difficult as we're volunteers, but knowing dependencies and critical path could be very helpful.

Can y'all please take a look and let me know of any comments and questions?

If we do this, next step would be to work together to build something that we all feel is reasonably OK.

**Comment by rtullybarr** [comment]
Hey! I have no thoughts on the electronics project plan, but I was wondering on the PC software side if we might want a higher-level diagram of some kind first (similar to the one you showed at our last meeting) before making a class diagram?
I think my current understanding would put these tasks under the 'PC Software' area as well (not sure if these are at the right granularity for this project plan):
- Define protocol for transferring data to PC (shared between Firmware and PC Software)
- Implement reading from USB HID device according to said protocol
- Implement EDF writing (and possibly reading)
- Main EEG UI: graphing waveforms (demo complete)
- Filtering / processing the data?
- Continuous writing of EDF file while capturing data?

I'm guessing some parts of the PC software have a dependency on the design (define workflow, UI wireframes)

**Comment by ustad-nordin** [comment]
Hi there,

This is a great idea and project, but I don't see any activities since 2022. Is this project discontinued? You're not like threatened by the big medical industry right?

**Reply by longevity-software** [reply]
Hi @ustad-nordin, I do not know if it is officially discontinued, but life definitely got in the project for a lot of us who were contributing.
